### ISSUE TITLE
[Back-End] Leverage the Lighthouse VA API to query VA forms

### ISSUE THREAD

**Issue by ncksllvn** [department-of-veterans-affairs/va.gov-team#3563]
The Lighthouse team has recently built an [API method](https://dev-api.va.gov/services/va_forms/v0/forms) for querying VA forms. This is perfect for us to leverage in our new Find a VA Form application. However, Lighthouse API methods require an `apiKey`, meaning we're not able to use it like we are other API methods. So, we need a solution here before we can begin leveraging it. Our options are -

1. Create an `apiKey` that the front-end can use during its build. Then, during the front-end build, we can use that to query the Lighthouse API method, then cache the response data into a static JSON file or write into the body of the HTML page.
    - Pros: No HTTP request from the browser; can be done entirely in vets-website
    - Cons: Data is only updated during front-end builds; may create a pattern that is only useful for this one case; adds additional time to our front-end build
1. Create an `apiKey` for the API itself. Then, we can create a new API method that signs the request, queries the Lighthouse API method, and returns the response data to the browser.
    - Pros: On the front-end, this is inline with how we use our API; no overhead added to our front-end build; data is refreshed without a front-end build; defines a standard pattern for cases in the future where we may want to use a Lighthouse API from www.va.gov
    - Cons: Creates work to do on the API, and our team is largely front-end engineers; kind of a weird pattern on the API

Option 2 was [proposed by Lighthouse engineers](https://dsva.slack.com/archives/C0MQ281DJ/p1574111390163200), so that is probably our best bet.

**Comment by ncksllvn** [comment]
@kelsonic What are your thoughts on this?

**Comment by ncksllvn** [comment]
I think the best approach here is option 2, with the most decisive reason being to avoid increasing our front-end deployment times, but also because this will help define a pattern for future work with integrating the front-end with a Lighthouse API.

**Comment by ncksllvn** [comment]
The source code for the Lighthouse API is [here](https://github.com/department-of-veterans-affairs/vets-api/tree/master/modules/va_forms). Hopefully, we'll be able to import and reuse most of that code for our purposes.

Essentially, the workflow will be -

1. Create an `apiKey` stored [here](https://github.com/department-of-veterans-affairs/devops)
1. Create a route in vets-api at `v0/va_forms`
1. That route should send a request to the Lighthouse API at `/services/va_forms/v0/forms`
    - This is important that we send an HTTP request, because the infrastructure from Lighthouse is different than the rest of vets-api.
1. Borrow as much code as possible for models, Swagger, etc.

**Comment by micahchiang** [comment]
Tagging the eBenefits engineers on here for visibility sake, since this is something that has some pretty significant ramifications for work we will be doing.

**Comment by ncksllvn** [comment]
I created a follow-up ticket for adding the secret API key into Credstash here, https://github.com/department-of-veterans-affairs/va.gov-team/issues/4279. I plan on closing this today after I merge my PR.

**Comment by ncksllvn** [comment]
Behold - https://dev-api.va.gov/v0/forms

**Comment by ncksllvn** [comment]
@micahchiang We added an API Key into Credstash that your team should be able to leverage to talk to other Lighthouse APIs. The ticket for that is here, https://github.com/department-of-veterans-affairs/va.gov-team/issues/4279.
